src/uu/ls/src/url_escaper.rs

```diff
@@ -0,0 +1,147 @@
+use std::{borrow::Cow, convert::{TryFrom, TryInto}, ffi::OsStr, ptr, mem::ManuallyDrop};
+
+mod byte_or_escape
+{
+    #[derive(Copy, Clone, PartialEq, Eq)]
+    #[repr(transparent)]
+    pub struct ByteOrEscape(u16);
```

Comment:
This feels like it should be an enum. Something like:
```rust
pub enum UrlChar {
    Byte(u8),
    Escaped(u8, u8)
}
```
Again, feels like premature optimization.

Reply:
That would not fit in 16 bits and thus result in a larger table (unless you use NonZeroU8 in the Escaped variant), as well as not enabling efficient counting of the number of bytes/escapes.

Reply:
This is only allocated once right? I don't feel like saving those few bytes is very important. Where is the counting important to this algorithm?